webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderUserDetailsFile.scala

```diff
@@ -212,6 +216,43 @@ final class FileUserDetailListProvider(roleApiMapping: RoleApiMapping, authorisa
   }
 
   override def authConfig: UserDetailList = cache
+
+  /**
+   * Filter the list of users by checking if the username
+   * is unique according to case sensitivity:
+   *
+   *   - if `case sensitivity` is enable
+   *     log a warn if there is potential collision when
+   *     this parameter will be disabled
+   *
+   *   - users with same username (according to the case sensitivity)
+   *     will be removed from the returned list
+   */
+  private[this] def sanityLoginFilter(userDetails: UserDetailList): UserDetailList = {
+    val userLogins = if(RudderConfig.rudderUsernameCaseSensitive){
+      userDetails.users.values.map(_.getUsername)
+    } else {
+      userDetails.users.values.map(_.getUsername.toLowerCase())
+    }
+    val duplicatesLogin = userLogins.groupBy(identity).collect { case (x, List(_,_,_*)) => x }.toList
+    println(duplicatesLogin)
+    println(userLogins)
+
+    (duplicatesLogin.isEmpty, RudderConfig.rudderUsernameCaseSensitive) match {
+      case (true, true) =>
+        val potentialDup = userLogins.map(_.toLowerCase()).groupBy(identity).collect { case (x, List(_,_,_*)) => x }
+        if(potentialDup.nonEmpty) {
+          ApplicationLogger.warn(s"Users with potential username collision if case sensitivity is deactivated: ${potentialDup.mkString(", ")}")
+          userDetails
+        } else {
+          userDetails
+        }
+      case (false, _) =>
```

Comment:
This will never happen here because one duplicate seems to be removed before, for exemple if I have 2 user `toto` it seems that one of them will be ignored